### PR TITLE
Adding support to build multi-arch Boundary images

### PR DESCRIPTION
**Pull request by jacobmammoliti**
- I've updated the release Dockerfile to support multiple architectures.

- I've also added a snippet in the Makefile called `docker-multiarch-build` that will build and push multiple architectures of the image to our Dockerhub. I've done it with Docker's `buildx` feature.

### CODE REVIEW

File: docker/Release.dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM docker.mirror.hashicorp.services/alpine:3.10
+FROM alpine:3.13
```
Comment (malnick) [inline docker/Release.dockerfile:1]: @mdeggies I think we need to stick to the hashicorp mirror for this image, is that correct? Do we have one for 3.13?

**Review by malnick — COMMENTED** [review]
Looks great! I need to check with @mdeggies on whether or not we have a 3.13 alpine image in our HashiCorp mirror. Going to relay this to her for final review.

File: docker/Release.dockerfile (hunk 1 above)
Comment (alvin-huang) [reply docker/Release.dockerfile:1]: Yes, this should be `docker.mirror.hashicorp.services/alpine:3.13`

Comment (jacobmammoliti) [reply docker/Release.dockerfile]: Cool. Updated now!

Comment (mdeggies) [reply docker/Release.dockerfile]: Just a note, in the future you can run `docker pull docker.mirror.hashicorp.servies/$whatever` locally to see if that image is supported in the docker mirror. If it's not, you can create a PR to add that here: https://github.com/hashicorp/docker-registry-mirror/blob/main/images.tf.

**Review by malnick — APPROVED** [review]

**Pull request merged**